frontend/src/components/profile/PrivacySettings.tsx

```diff
@@ -0,0 +1,272 @@
+import * as React from "react"
+import { Grid, Button, Paper, Typography } from "@material-ui/core"
+import FormControlLabel from "@material-ui/core/FormControlLabel"
+import Radio from "@material-ui/core/Radio"
+import RadioGroup from "@material-ui/core/RadioGroup"
+import { makeStyles } from "@material-ui/core/styles"
+import { request } from "../../utils/fetch"
+import type { PrivacySetters } from "../../utils/profile"
+import { getPrivacySettingsUpdates } from "../../utils/profile"
+
+type PrivacySettingsProps = {
+  readonlySettings: PrivacySettingsType
+  editableSettings: PrivacySettingsType
+  privacySetters: PrivacySetters
+  setPrivacySettings: React.Dispatch<React.SetStateAction<PrivacySettingsType>>
+  userId: number
+}
+
+// A user's privacy settings.
+export type PrivacySettingsType = {
+  first_name_privacy: PrivacyKey
+  last_name_privacy: PrivacyKey
+  birth_date_privacy: PrivacyKey
+  gender_privacy: PrivacyKey
+  interests_privacy: PrivacyKey
+}
+
+export type PrivacyKey = "" | "PU" | "PR" | "HI"
+
+export type PrivacyValue = "Public" | "Private" | "Hidden"
+
+type PrivacyChoice = {
+  [key: string]: PrivacyValue
+}
+
+export const PRIVACY_CHOICES: PrivacyChoice = {
+  PU: "Public",
+  PR: "Private",
+  HI: "Hidden",
+}
+
+// A field listed in the settings.
+type PrivacyField = {
+  title: string
+  privacy_level: PrivacyKey
+  setPrivacyLevel: (value: PrivacyKey) => void
+}
+
+const useStyles = makeStyles({
+  pageTitle: {
+    fontSize: "2rem",
+    fontWeight: "bold",
+  },
+  fieldsContainer: {
+    paddingTop: 40,
+    paddingBottom: 10,
```

Comment:
No unitless css please. Yes it defaults to px, but I'd prefer we be explicit. Also I'd prefer rem

Reply:
lots of pages and components currently use the default px unit. whatever we choose it should be consistent throughout the app

Reply:
We are choosing to have units on all css.

Reply:
do you want to add that to this pr or create another one for it

Reply:
For the files in this PR, change it. We can fix others as they come up